core/services/ocr2/plugins/ocr2keeper/evm21/upkeepstate/store.go

```diff
@@ -168,18 +217,44 @@ func (u *upkeepStateStore) fetchPerformed(ctx context.Context, start, end int64,
 	for _, workID := range performed {
 		if _, ok := u.cache[workID]; !ok {
 			s := &upkeepStateRecord{
-				workID:  workID,
-				state:   ocr2keepers.Performed,
-				addedAt: time.Now(),
-				block:   uint64(end), // TODO: use block number from log
+				WorkID:          workID,
+				CompletionState: ocr2keepers.Performed,
+				AddedAt:         time.Now(),
+				BlockNumber:     uint64(end), // TODO: use block number from log
 			}
+
 			u.cache[workID] = s
 		}
 	}
 
 	return nil
 }
 
+// fetchFromDB fetches all upkeeps indicated as ineligible from the db to
+// populate the cache
+func (u *upkeepStateStore) fetchFromDB(ctx context.Context, workIDs ...string) error {
+	states, err := u.orm.SelectStatesByWorkIDs(workIDs)
```

Comment:
Why is the context not passed?

Reply:
good call. i'll do that with that one and the other.